tests/rest/client/v2_alpha/test_account.py

```diff
@@ -475,6 +476,23 @@ def test_GET_whoami(self):
         whoami = self.whoami(tok)
         self.assertEqual(whoami, {"user_id": user_id, "device_id": device_id})
 
+    def test_GET_whoami_appservices(self):
+        user_id = "@as_user_kermit:test"
+        as_token = "i_am_an_app_service"
+
+        appservice = ApplicationService(
+            as_token,
+            self.hs.config.server_name,
+            id="1234",
+            namespaces={"users": [{"regex": r"@as_user.*", "exclusive": True}]},
```

Comment:
Why do we need to treat `@as_user.*` as a raw string rather than a normal one?

Reply:
copy/paste? :P 

Can probably just use the same variable to do that though: will make the change and merge.